### ISSUE TITLE
Modal close button not a11y ready

### ISSUE THREAD

**Issue by HardeepAsrani** [Codeinwp/otter-blocks#2714]
Reported: https://secure.helpscout.net/conversation/3130832256/475624

The Close button of Modal block lacks aria-labels and others to help users with screen-readers.
